nevow/test/test_appserver.py

```diff
@@ -292,3 +330,145 @@ def test_emptyPostPath(self):
             (childResource, ()), request, ('foo', 'bar'), context)
         self.assertEqual(request.prepath, [''])
         self.assertEqual(request.postpath, [])
+
+
+class OldResourceAdapterTests(TestCase):
+    def test_deferredResource(self):
+        expected = b"success"
+        resource = Data(expected, b"text/plain")
+        deferredResource = DeferredResource(succeed(resource))
+
+        rootPage = Page()
+        rootPage.child_deferredresource = deferredResource
+
+        actual = self.successResultOf(
+            renderResourceReturnTransport(
+                rootPage,
+                b"/deferredresource",
+                b"GET",
+            ),
+        )
+        self.assertTrue(
+            actual.endswith(b"\r\n\r\n" + expected),
+            "{!r} did not include expected response body {!r}".format(
+                actual,
+                expected,
+            )
+        )
+
+    def test_deferredResourceChild(self):
+        expected = b"success"
+        resource = Data(expected, b"text/plain")
+        intermediate = Data(b"incorrect intermediate", b"text/plain")
+        intermediate.putChild(b"child", resource)
+        deferredResource = DeferredResource(succeed(intermediate))
+
+        rootPage = Page()
+        rootPage.child_deferredresource = deferredResource
+
+        actual = self.successResultOf(
+            renderResourceReturnTransport(
+                rootPage,
+                b"/deferredresource/child",
+                b"GET",
+            ),
+        )
+        self.assertTrue(
+            actual.endswith(b"\r\n\r\n" + expected),
+            "{!r} did not include expected response body {!r}".format(
+                actual,
+                expected,
+            )
+        )
+
+@implementer(IRealm)
+class Realm(object):
```

Comment:
Could you name this a bit less ambiguously, `OneIResourceAvatarRealm` or somesuch?